### PR TITLE
Horizon 30 support

### PR DESCRIPTION
**Pull request by deejgregor**
This builds on #5.

### CODE REVIEW

**Comment by agalue** [comment]
Looks good, but I've been told that starting from Horizon 31, the installation will be based on tarball like Minion (instead of `apt` or `dpkg`).

We can keep the logic as is or make assumptions about compatibility.

The package manager logic is to decide if the chosen version supports the Twin API for Kafka. We can remove that logic and explicitly say that the Helm chart works **only** with Horizon 29 (or newer) or Meridian 2022 (or newer).

I wrote this last year, so Meridian 2021 (the latest version at that time) didn't support Twin API; therefore, the logic. Also, Meridian was initially considered, but I believe the customer is not using it due to the extra work of building and maintaining images (even if details about that are on the README; perhaps that won't work anymore).

**Comment by deejgregor** [comment]
> Looks good, but I've been told that starting from Horizon 31, the installation will be based on tarball like Minion (instead of `apt` or `dpkg`).

Not just in Horizon 31. It turns out it was back-ported and is in 30.0.4, and I ran squarely and quite firmly into it. :)

Here's what I did: https://github.com/opennms-forge/onms-k8s-poc/pull/6/commits/ff2fac746bc45dcaf0717515fd46324f03ef9e8b

File: opennms/values.yaml [hunk 1]
```diff
@@ -105,10 +105,10 @@ opennms:
     - RRA:AVERAGE:0.5:288:366
     - RRA:MAX:0.5:288:366
     - RRA:MIN:0.5:288:366
-    alecImage:
-      repository: opennms/alec
-      pullPolicy: IfNotPresent
-      tag: v1.1.1
+    alecImage: # If alecImage is specified, the KAR is retrieved from the specified Docker image used as an initContainer, otherwise the latest release in GitHub is retrieved
```
Comment (agalue) [inline opennms/values.yaml]: To be safe, I think we should set it as `alecImage: {}` when there is no content within it and let the chart knows that it's a Map.

Comment (deejgregor) [reply opennms/values.yaml]: +1. Done.

**Comment by agalue** [comment]
Looks good, but as mentioned, the following won't work:
https://github.com/opennms-forge/onms-k8s-poc/blob/horizon-30/opennms/scripts/onms-core-init.sh#L84-L94

When using `version.properties`, you don't know if we're talking about Meridian or Horizon, so the logic will *always* take the `else` route.

That's why I suggested removing that logic altogether and setting new minimum requirements for the whole chart supporting the Twin API regardless of the OpenNMS flavor.

Alternatively, we can infer the flavor based on the version number scheme. For instance, the version for Meridian always starts with `2019` or `2021`, and so on, whereas for Horizon would be `29`, `30`, etc. I believe we won't be alive when those overlap ;)

**Comment by deejgregor** [comment]
> Alternatively, we can infer the flavor based on the version number scheme. For instance, the version for Meridian always starts with `2019` or `2021`, and so on, whereas for Horizon would be `29`, `30`, etc. I believe we won't be alive when those overlap ;)

I was aiming to fix that in OpenNMS so it included the flavor, but it was easy to do this now: https://github.com/opennms-forge/onms-k8s-poc/pull/6/commits/7faa8882ef7a119c1bd187012d43653adc2a1d81

**Review by agalue — APPROVED** [review]
LGTM! I can review Loki after, now that I'm well familiar with all its flavors and issues after the refactorization they made on their Helm Chart (which I'm using on other projects in all possible ways).

**Pull request merged**